### PR TITLE
Use third-party topological sort

### PR DESCRIPTION
**Pull request by knoepfel**
The DE framework has been using a homegrown topological sorting algorithm to detect cycles between consumed and produced products.  This PR replaces the homegrown version with that provided by the `toposort` module.

- No new dependencies are added with this PR.
- This PR additionally checks that the products consumed by publishers are also produced upstream.
- No checking is yet done for products consumed by the logic-engine via the facts and rule expressions.

### CODE REVIEW

File: src/decisionengine/framework/config/ChannelConfigHandler.py [hunk 1]
```diff
@@ -66,67 +66,50 @@ def _validate(channel):
...
         except AttributeError as msg:
             raise RuntimeError(f"transform module {tname} does not have required lists {msg}")
 
+    # Publishers
+    for pname, conf in channel.get('publishers').items():
+        my_module = importlib.import_module(conf.get('module'))
+        try:
+            consumed_products[pname] = set(getattr(my_module, 'CONSUMES'))
+        except AttributeError:
+            raise RuntimeError(f"publisher module {pname} does not have required CONSUMES list")
```
Comment (jcpunk) [inline src/decisionengine/framework/config/ChannelConfigHandler.py:95]: Can this be covered with a unit test?

File: src/decisionengine/framework/config/ChannelConfigHandler.py [hunk 2]
```diff
@@ -66,67 +66,50 @@ def _validate(channel):
...
-    if cyclic_modules:
-        raise RuntimeError(f"cyclic dependency detected for modules {list(cyclic_modules)}")
+    for consumer, products in consumed_products.items():
+        graph[consumer] = map(lambda p: produced_products.get(p), products)
+
+    # Do the check
+    try:
+        toposort.toposort(graph)
+    except Exception as e:
+        raise RuntimeError(f"A produces/consumes circularity exists in the configuration:\n{e}")
```
Comment (jcpunk) [inline src/decisionengine/framework/config/ChannelConfigHandler.py:112]: Can this be covered with a unit test?

File: src/decisionengine/framework/config/ChannelConfigHandler.py (hunk 1 above)
Comment (knoepfel) [reply src/decisionengine/framework/config/ChannelConfigHandler.py]: Done.

File: src/decisionengine/framework/config/ChannelConfigHandler.py (hunk 2 above)
Comment (knoepfel) [reply src/decisionengine/framework/config/ChannelConfigHandler.py:120]: Done.

**Comment by knoepfel** [comment]
@jcpunk, I believe I have addressed your concerns.

**Review by jcpunk — APPROVED** [review]
+1 all looks good to me

**Pull request merged**